### PR TITLE
Support scoped configuration scan summary types

### PR DESCRIPTION
**Pull request by Daniel-GrunbergerCA**
Support `configurationscansummaries` objects for namespace and cluster scope:

1. ` kubectl get configurationscansummaries.spdx.softwarecomposition.kubescape.io <namespace>` - This will return a `ConfigurationScanSummary` object for the given namespace. It will be build based on the `WorkloadConfigurationScanSummary` objects that exist for that namespace
2. ` kubectl get configurationscansummaries.spdx.softwarecomposition.kubescape.io` - This will return a `ConfigurationScanSummaryList` object, where each item is a summary for a namespace.

### CODE REVIEW

**Review by vladklokun — APPROVED** [review]
Type definitions and registrations look good to me.

File: pkg/registry/file/configurationscansummarystorage.go [hunk 1]
```diff
@@ -0,0 +1,211 @@
...
+		return err
+	}
+
+	if workloadScanSummaryListObjPtr == nil {
+		return storage.NewInternalError("workload scan summary list is nil")
+	}
+
+	if len(workloadScanSummaryListObjPtr.Items) == 0 {
+		return storage.NewKeyNotFoundError(key, 0)
+	}
```
[the remark this answers is not in the document]
Comment (Daniel-GrunbergerCA) [reply pkg/registry/file/configurationscansummarystorage.go:85]: We aren't sure how to create proper custom errors, current behavior of key not found is acceptable, so skipping it for now

File: pkg/registry/file/configurationscansummarystorage.go [hunk 2]
```diff
@@ -0,0 +1,211 @@
...
+
+	configurationScanSummaryObj := buildConfigurationScanSummary(*workloadScanSummaryListObjPtr, namespace)
```
[the remark this answers is not in the document]
Comment (Daniel-GrunbergerCA) [reply pkg/registry/file/configurationscansummarystorage.go:87]: thanks :)

**Review by vladklokun — CHANGES_REQUESTED** [review]
Let’s make the changes we discussed, and this is good to go. Good job!

**Review by vladklokun — APPROVED** [review]
Change the OS filesystem to an in-memory one and we’re good to go. Thank you for your hard work! :)

**Review by vladklokun — APPROVED** [review]
Please remember to make a meaningful merge commit message.

**Pull request merged**